File: src/test/java/codeExamples/libUsages/java/example4/LibUsage.java
package codeExamples.libUsages.java.example4;

import codeExamples.libExamples.java.example4.*;

public class LibUsage {

    public static void main(String[] args) {
        for (int i = 0; i < 10; i++) {
            use(i);
        }
    }

    public static void use(int n) {
        Class1.f4();
        if (n % 2 == 0) {
            Class1.f3();
        }
        Class1 inst1 = new Class1();
        inst1.f1();
        for (int i = 0; i < n; i++) {
            inst1.f2();
        }
    }
}

File: src/test/kotlin/codeExamples/libUsages/kotlin/example2/LibUsage.kt
package codeExamples.libUsages.kotlin.example2

import codeExamples.libExamples.kotlin.example2.Class1
import codeExamples.libExamples.kotlin.example2.Class2

fun use1(n: Int) {
    val inst1 = Class1("inst1")
    val inst2 = Class2(2)
    println(inst1.limit)
    for (i in 0 until n) {
        inst1.incrementI()
        inst1.incrementI(8)
    }
    if (n % 2 == 0)
        inst1.writeIToS()
    else
        inst1.setI(7)
    inst1.incrementI(inst2)
    inst1.writeIToS()
    inst1.getS()
    inst1.setIToNull()
    inst1.setI(6)
}

fun main() {
    for (n in 0 until 5) {
        use1(n)
    }
}